internal/logic/logic_test.go

```diff
@@ -319,10 +321,282 @@ func TestTransactionManagementServiceLogic_NewTransaction(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			rec := NewTransactionManagementServiceLogic(tt.setup())
-			time.Sleep(2 * time.Second)
+
 			got := rec.NewTransaction(tt.credentials)
 
 			tt.want(got)
 		})
 	}
 }
+
+func TestTransactionManagementServiceLogic_DownloadTransactions(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	tests := []struct {
+		name        string
+		credentials string
+		setup       func() (datasource.DataSourceI, config.ExternalSvc)
+		want        func(*respModel.Response)
+	}{
+		{
+			name:        "Success :: DownloadPdf",
+			credentials: "123",
+			setup: func() (datasource.DataSourceI, config.ExternalSvc) {
+				mockDs := mock.NewMockDataSourceI(mockCtrl)
+				var trans []model.Transaction
+				trans = append(trans, model.Transaction{UserId: "123", AccountNumber: 1})
+				mockDs.EXPECT().Get(map[string]interface{}{"transaction_id": "123"}, 0, 0).Times(1).Return(trans, 1, nil)
+				router := mux.NewRouter()
+				router.HandleFunc("/microbank/v1/user", func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusOK, "Success", map[string]interface{}{"Name": "abc"})
```

Comment:
All fields that are passed to the mocked service in the incoming request must be validated.

Reply:
done

Reply:
Still seeing cases where the validation is bot done. Update.

Reply:
Still seeing cases where the validation is not done. Update.

Reply:
as it is a get request so I am only passing cookie to it and it has name and value so I have added the validation for them